### PR TITLE
Move opentelemetry-exporter-otlp to opentelemetry. Remove opentelemetry-exporter-jaeger

### PR DESCRIPTION
**Pull request by brunobat**
Part of #26444 and https://issues.redhat.com/browse/QUARKUS-2394
Depends on https://github.com/quarkiverse/quarkus-opentelemetry-exporter/pull/1 (merged but not released)
Remove quarkus-opentelemetry-parent-aggregator and ...-exporter-jaeger 
Move ...-exporter-otlp inside quarkus-opentelemetry as default exporter
Update Docs

This PR introduces breaking changes

### CODE REVIEW

**Comment by brunobat** [comment]
Draft because it cannot be merged before https://github.com/quarkiverse/quarkus-opentelemetry-exporter/pull/1

**Comment by rsvoboda** [comment]
Don't forget to add relocation rule (https://github.com/quarkusio/quarkus/tree/main/relocations) to point to https://github.com/quarkiverse/quarkus-opentelemetry-exporter

**Comment by radcortez** [comment]
Let's check with @gsmet to make sure if we are following all the required steps to move the extension to Quarkiverse.

File: docs/src/main/asciidoc/opentelemetry.adoc [hunk 1]
```diff
@@ -431,8 +431,13 @@ Message out = Message.of(...).withMetadata(tm);
 The above creates a `TracingMetadata` object we can add to the `Message` being produced,
 which retrieves the OpenTelemetry `Context` to extract the current span for propagation.
 
+== Exporters
+
+Quarkus OpenTelemetry defaults to the standard OTLP exporter defined in OpenTelemetry.
+
+Additional exporters will be available in Quarkiverse at https://github.com/quarkiverse/quarkus-opentelemetry-exporter/blob/main/README.md[quarkus-opentelemetry-exporter] project.
```
Comment (gsmet) [inline docs/src/main/asciidoc/opentelemetry.adoc:438]:
```suggestion
Additional exporters will be available in the Quarkiverse https://github.com/quarkiverse/quarkus-opentelemetry-exporter/blob/main/README.md[quarkus-opentelemetry-exporter] project.
```

File: relocations/pom.xml [hunk 2]
```diff
@@ -57,6 +57,9 @@
         <module>quarkus-logging-sentry-deployment</module>
         <module>quarkus-neo4j</module>
         <module>quarkus-neo4j-deployment</module>
+        <module>quarkus-opentelemetry-exporter-jaeger</module>
```
Comment (gsmet) [inline relocations/pom.xml:60]: You missed `quarkus-opentelemetry-exporter-jaeger-deployment` here.

**Review by gsmet — CHANGES_REQUESTED** [review]
I spotted a few very minor things but this is extremely clean. Thanks.

**Comment by gsmet** [comment]
BTW, at least some of the CI failures look related though so you should have a look at them too.

**Comment by gsmet** [comment]
@brunobat when you next adjust this PR, could you make the commit message more explicit. It is a bit confusing now.

**Comment by brunobat** [comment]
Yes I will @gsmet

**Comment by brunobat** [comment]
@gsmet there was a missing runtime dependency on opentelemetry: quarkus-grpc-common.
Also updated the commit message.

**Review by radcortez — APPROVED** [review]

File: relocations/quarkus-opentelemetry-exporter-otlp-deployment/pom.xml [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+    </parent>
+    <modelVersion>4.0.0</modelVersion>
+
+    <artifactId>quarkus-opentelemetry-exporter-otlp-deployment</artifactId>
+
+    <distributionManagement>
+        <relocation>
+            <groupId>io.quarkus</groupId>
+            <artifactId>quarkus-opentelemetry-deployment</artifactId>
+            <version>2.14.0.Final</version>
```
Comment (gsmet) [inline relocations/quarkus-opentelemetry-exporter-otlp-deployment/pom.xml:18]:
```suggestion
            <version>${project.version}</version>
```

File: relocations/quarkus-opentelemetry-exporter-otlp/pom.xml [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+    </parent>
+    <modelVersion>4.0.0</modelVersion>
+
+    <artifactId>quarkus-opentelemetry-exporter-otlp</artifactId>
+
+    <distributionManagement>
+        <relocation>
+            <groupId>io.quarkus</groupId>
+            <artifactId>quarkus-opentelemetry</artifactId>
+            <version>2.14.0.Final</version>
```
Comment (gsmet) [inline relocations/quarkus-opentelemetry-exporter-otlp/pom.xml:18]:
```suggestion
            <version>${project.version}</version>
```

File: bom/application/pom.xml [hunk 5]
```diff
@@ -2780,26 +2780,6 @@
...
-            <dependency>
-                <groupId>io.quarkus</groupId>
-                <artifactId>quarkus-opentelemetry-exporter-otlp-deployment</artifactId>
-                <version>${project.version}</version>
-            </dependency>
-            <dependency>
-                <groupId>io.quarkus</groupId>
-                <artifactId>quarkus-opentelemetry-exporter-otlp</artifactId>
-                <version>${project.version}</version>
-            </dependency>
```
Comment (gsmet) [inline bom/application/pom.xml:2802]: You shouldn't remove them. You need to move them to the Relocations section at the bottom.

See https://github.com/quarkusio/quarkus/blob/main/bom/application/pom.xml#L5855

**Review by gsmet — CHANGES_REQUESTED** [review]
Sorry I didn't spot them earlier but I spotted 3 small issues regarding the relocations:
- don't remove them from the BOM, otherwise projects will fail as they didn't specify the version before (that's why the quickstart compilation is failing)
- you shouldn't point to a static version for the artifacts that are still in the core, otherwise you will end up with something a bit inconsistent.

Apart from that, I think we're good.

File: bom/application/pom.xml (hunk 5 above)
Comment (brunobat) [reply bom/application/pom.xml:2802]: Done, thanks for spotting it.

**Review by gsmet — APPROVED** [review]

**Comment by gsmet** [comment]
I'll fix this PR tomorrow. We need to adjust the CI run.

**Pull request merged**